### PR TITLE
feat: new privacy settings for orgs

### LINKED ISSUE #13691: [CAL-3143] "Privacy" tab for Orgs
We need more privacy settings, introducing a new "Privacy" tab

* hide all teams on [acme.cal.com](http://acme.cal.com/)
* hide all event-types on teams: [acme.cal.com/sales](http://acme.cal.com/sales)
* hide all members on teams: [acme.cal.com/sales](http://acme.cal.com/sales)

we should move this component in there too:

image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3143](https://linear.app/calcom/issue/CAL-3143/privacy-tab-for-orgs)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
CleanShot 2024-02-14 at 19 00 55@2x

### CODE REVIEW

File: packages/features/ee/teams/components/MakeTeamPrivateSwitch.tsx [hunk 1]
```diff
@@ -46,6 +46,52 @@ const MakeTeamPrivateSwitch = ({
...
+        <>
+          <SettingsToggle
+            toggleSwitchAtTheEnd={true}
+            title={t("hide_all_teams")}
+            disabled={disabled || mutation?.isPending}
+            description={t("hide_all_teams_description")}
+            checked={isTeamPrivate}
+            onCheckedChange={(checked) => {
+              setTeamPrivate(checked);
+              mutation.mutate({ id: teamId, isPrivate: checked });
```
Comment (PeerRich) [inline packages/features/ee/teams/components/MakeTeamPrivateSwitch.tsx:60]: need help here. can be a follow up PR

Comment (Udit-takkar) [reply packages/features/ee/teams/components/MakeTeamPrivateSwitch.tsx:60]: @PeerRich I think i already implemented hiding members for team. if you check the 'make team private' switch then you won't be able to see members

Screenshot 2024-02-22 at 1 26 35 AM

Comment (Udit-takkar) [reply packages/features/ee/teams/components/MakeTeamPrivateSwitch.tsx:60]: should i create separate switches for 'hiding event types', 'hiding members' and 'hiding team in orgs'

**Comment by Udit-takkar** [comment]
Just pushed some code to hide teams and team members of org if the org is private and as far as event types are concerned. We already have this switch to hide event types

Screenshot 2024-02-22 at 2 14 25 AM

**Review by CarinaWolli — DISMISSED** [review]
I think we also shouldn't show the team members of a team if the org is private.

**Comment by Udit-takkar** [comment]
> I think we also shouldn't show the team members of a team if the org is private.

Okay done.

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -1083,6 +1083,7 @@
   "make_team_private": "Make team private",
   "make_team_private_description": "Your team members won't be able to see other team members when this is turned on.",
   "you_cannot_see_team_members": "You cannot see all the team members of a private team.",
+  "you_cannot_see_teams_of_org":"You cannot see teams of a private org.",
```
Comment (CarinaWolli) [inline apps/web/public/static/locales/en/common.json:1086]:
```suggestion
  "you_cannot_see_teams_of_org":"You cannot see teams of a private organization.",
```

File: packages/features/ee/organizations/pages/settings/members.tsx [hunk 3]
```diff
@@ -11,29 +10,24 @@ import { Meta } from "@calcom/ui";
...
-            teamId={currentOrg.id}
-            isPrivate={currentOrg.isPrivate}
-            disabled={isInviteOpen}
-          />
-        )}
-      </div>
+      <div>{!isPending && canLoggedInUserSeeMembers && <UserListTable />}</div>
+      {!canLoggedInUserSeeMembers && (
+        <div className="border-subtle rounded-xl border p-6">
+          <h2 className="text-default">{t("only_admin_can_see_members_of_org")}.</h2>
```
Comment (CarinaWolli) [inline packages/features/ee/organizations/pages/settings/members.tsx:28]:
```suggestion
          <h2 className="text-default">{t("only_admin_can_see_members_of_org")}</h2>
```

File: packages/features/ee/teams/components/MakeTeamPrivateSwitch.tsx [hunk 4]
```diff
@@ -46,6 +46,52 @@ const MakeTeamPrivateSwitch = ({
         switchContainerClassName="mt-6"
         data-testid="make-team-private-check"
       />
+      {/* TODO: add mutations and logic
```
Comment (CarinaWolli) [inline packages/features/ee/teams/components/MakeTeamPrivateSwitch.tsx:49]: can this be removed?

**Review by CarinaWolli — COMMENTED** [review]
I fixed a type error but there are more, @Udit-takkar can you fix them? Added some minor clean up comments, but other than that it looks good 🙏

**Review by CarinaWolli — APPROVED** [review]
Looks good 🙏

**Comment by PeerRich** [comment]
thank you @Udit-takkar for taking this over 🙏

**Comment by PeerRich** [comment]
fixes #13421

**Pull request merged**